### PR TITLE
notebooks: add stars to the GQL API

### PR DESCRIPTION
**Pull request by novoselrok**
This PR adds the ability to create and delete stars for notebooks as well as list and count them.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/notebooks.graphql [hunk 1]
```diff
@@ -206,6 +220,55 @@ type Notebook implements Node {
...
+}
+
+"""
+A notebook star.
+"""
+type NotebookStar {
+    """
+    User that starred the notebook.
+    """
+    user: User
```
Comment (camdencheek) [inline cmd/frontend/graphqlbackend/notebooks.graphql:267]: Should this be `User!`?

File: enterprise/cmd/frontend/internal/notebooks/resolvers/resolvers.go [hunk 2]
```diff
@@ -356,6 +365,15 @@ func (r *notebookResolver) ViewerCanManage(ctx context.Context) bool {
 	return validateNotebookWritePermissionsForUser(r.notebook, actor.UID) == nil
 }
 
+func (r *notebookResolver) ViewerHasStarred(ctx context.Context) bool {
+	user, err := r.db.Users().GetByCurrentAuthUser(ctx)
+	if err != nil {
+		return false
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/notebooks/resolvers/resolvers.go:371]: Why not just return the error here? Seems like swallowing the error could make issues difficult to debug

Comment (novoselrok) [reply enterprise/cmd/frontend/internal/notebooks/resolvers/resolvers.go:371]: I guess I could change the return signature to `(bool, error)` and return the error alongside the bool value?

Comment (camdencheek) [reply enterprise/cmd/frontend/internal/notebooks/resolvers/resolvers.go:371]: I think that's probably better. The graphql layer will interpret that error nicely. We shouldn't ever have to call this manually, so the bonus error return value doesn't really add complexity

**Pull request merged**